openverse_catalog/dags/providers/provider_api_scripts/finnish_museums.py

```diff
@@ -42,54 +48,155 @@ class FinnishMuseumsDataIngester(ProviderDataIngester):
+
+        Most runs of the DAG will have a very small (or zero) number of records, so in
+        order to determine how many time intervals we need we first check the record
+        count, and split into the smallest number of intervals possible.
+
+        If the building has no/few records, this results in ONE extra request.
+        If the building has a large amount of data, this results in TWENTY FIVE extra
+        requests (one for the full day, and then one for each hour).
+        """
+        pairs_list: list[tuple[datetime, datetime]] = []
+        # Get UTC timestamps for the start and end of the ingestion date
+        start_ts = datetime.strptime(self.date, "%Y-%m-%d").replace(tzinfo=timezone.utc)
+        end_ts = start_ts + timedelta(days=1)
+
+        # TODO: The 10k and 100k threshholds are somewhat random. Test to see how they
+        # perform, probably move to constants. Should they be overridable?
+        # If this gets pulled out into a shared util that can be used by other DAGs,
+        # we should definitely at least make it so each DAG can provide its own
+        # threshold value.
+
+        record_count = self._get_record_count(start_ts, end_ts, building)
+        if record_count == 0:
+            logger.info(f"No data for {building}. Continuing.")
+            return pairs_list
+        if record_count < 10_000:
+            # This building only has a small amount of data. We can ingest all of the
+            # data for it in a single run.
+            return self._get_timestamp_query_params_list(start_ts, end_ts, 1)
+
+        # If we got here, this building has a large amount of data. Since data is often
+        # not distributed evenly across the day, we'll break down each hour of the day
+        # separately. We add timestamp_pairs to the list only for hours that actually
+        # contain data. Hours that contain more data get divided into a larger number of
+        # portions.
+        hour_slices = self._get_timestamp_query_params_list(start_ts, end_ts, 24)
+        for (start_hour, end_hour) in hour_slices:
+            # Get the number of records in this hour interval
+            record_count = self._get_record_count(start_hour, end_hour, building)
+            if record_count == 0:
+                # No records for this hour, don't bother ingesting for this time period.
+                logger.info(f"No data detected for {start_hour}. Continuing.")
+                continue
+            if record_count < 10_000:
+                # This hour doesn't have much data, ingest it in one chunk
+                pairs_list.append((start_hour, end_hour))
+                continue
+            # If we got this far, this hour has a lot of data. It is split into 12 5-min
+            # intervals if it has fewer than 100k, or 20 3-min intervals if more.
+            num_divisions = 12 if record_count < 100_000 else 20
+            min_slices = self._get_timestamp_query_params_list(
```

Comment:
reminder to myself to change the name `min_slices`, currently looks like 'minimum'